shop/src/test/java/ro/msg/learning/shop/service/impl/OrderServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.StockService;
+import ro.msg.learning.shop.service.impl.CustomerServiceImpl;
+import ro.msg.learning.shop.service.impl.LocationServiceImpl;
+import ro.msg.learning.shop.service.impl.OrderDetailServiceImpl;
+import ro.msg.learning.shop.service.impl.OrderServiceImpl;
+import ro.msg.learning.shop.service.impl.strategy.SingleLocationStrategy;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+@TestPropertySource(locations="classpath:application.properties")
+public class OrderServiceImplTest {
+
+    private static final String addressCountry = "Romania";
+    private static final String addressCity = "Cluj-Napoca";
+    private static final String addressCounty = "Cluj";
+    private static final String addressStreet = "strada Zorilor";
+
+    private static final String muresCounty = "Mures";
+
+    private static final String muresCity = "Targu Mures";
+    @Mock
+    private CustomerRepository customerRepository;
+
+    private StrategyConfiguration configuration;
+
+    @Mock
+    private SingleLocationStrategy singleLocationStrategy;
+    @Mock
+    private StockRepository stockRepository;
+    @Mock
+    private StockService stockService;
+    @Mock
+    private LocationServiceImpl locationService;
+    @Mock
+    private ProductOrderRepository productOrderRepository;
+
+    @Mock
+    private OrderDetailsRepository orderDetailsRepository;
+
+    @InjectMocks
+    private CustomerServiceImpl customerService;
+
+    @InjectMocks
+    private OrderDetailServiceImpl orderDetailService;
+
+    @InjectMocks
+    private OrderServiceImpl orderService;
+    private Location location1;
+    private Location location2;
+    private Customer customer;
+    private ProductOrder order;
+    private ProductCategory category;
+    private Supplier supplier1;
+    private Supplier supplier2;
+    private Product product1;
+    private Product product2;
+    private Stock stock1Cluj;
+    private Stock stock2Cluj;
+    private Stock stock1Mures;
+    private Stock stock2Mures;
+
+    private List<OrderDetail> orderDetails;
+    
+    @BeforeEach
+    void initialiseData() {
+        this.location1 = new Location(1, "Cluj-Napoca Location", addressCountry, addressCity, addressCounty, addressStreet);
+        this.location2 = new Location(2, "Targu Mures Location", addressCountry, muresCity, muresCounty, addressStreet);
+        this.customer = Customer.builder().id(2).email("raul.calugar@yahoo.ro").build();
+        this.order = ProductOrder.builder().addressCountry(addressCountry).addressCounty(addressCounty).addressCity(addressCity).addressStreetAddress(addressStreet).createdAt(LocalDateTime.now()).build();
+        this.category = ProductCategory.builder().id(1).name("IT").description("IT Products").build();
+        this.supplier1 = Supplier.builder().id(1).name("HP").build();
```

Comment:
names like location1, location2 and so on lead to names like "finalProject", "finalFinalProject", "lastFinalProject2022" and so on. Rename them into smth coherent like locationCj and locationTg.
Same goes for supplier, stock and other things that are enumerated in the name